onnxruntime/core/providers/webgpu/shader_helper.h

```diff
@@ -101,6 +101,10 @@ class ShaderHelper final {
     return *this;
   }
 
+  const std::vector<std::unique_ptr<ShaderVariableHelper>>& GetInputVars() {
```

Comment:
The reason why I do not want to expose the `input_vars_` directly, is because this getter exposed the internal implementation details and made the type as a part of public interface.

I would do it a little differently in concat.cc:
```
  std::vector<const ShaderVariableHelper *> input_vars;
  for (size_t i = 0; i < input_count_; ++i) {
    input_vars.push_back(&shader.AddInput("input_" + std::to_string(i), ShaderUsage::UseUniform | ShaderUsage::UseIndicesTypeAlias));
  }
```